packages/fast-tooling/src/web-components/units-text-field/units-text-field.ts

```diff
@@ -0,0 +1,117 @@
+        this.type = TextFieldType.text;
+        super.connectedCallback();
+
+        this.control.addEventListener("keydown", this.handleKeyDown);
+    }
+
+    /**
+     * Finds the first index of a regular expression in a string.
+     * @param searchString The string to search.
+     * @param regex The regular expression to match.
+     * @param position An optional starting position.
+     * @returns The first matching index within the string or -1 if it is not found.
+     */
+    public indexOf(searchString: string, regex: RegExp, position?: number) {
+        const str = position ? searchString.substring(position) : searchString;
+        const match = str.match(regex);
+        return match ? str.indexOf(match[0]) + position : -1;
+    }
+
+    /**
+     * Finds the last index of a regular expression in a string.
+     * @param searchString The string to search.
+     * @param regex The regular expression to match.
+     * @param position An optional starting position.
+     * @returns The last matching index within the string or -1 if it is not found.
+     */
+    public lastIndexOf(searchString: string, regex: RegExp, position?: number) {
+        const str = position ? searchString.substring(0, position) : searchString;
+        const match = str.match(regex);
+        return match ? str.lastIndexOf(match[match.length - 1]) : -1;
+    }
+
+    public handleKeyDown = (ev: KeyboardEvent): boolean => {
+        if (ev.key === keyArrowUp || ev.key === keyArrowDown) {
+            const amount: number =
+                (ev.shiftKey ? 10 : 1) * (ev.key === keyArrowUp ? 1 : -1);
+            const startPos: number = this.control.selectionStart;
+            const endPos: number = this.control.selectionEnd;
+            const origValue: string = this.control.value;
+            const isSelected: boolean = startPos !== endPos;
+
+            let replaceText: string = "";
+
+            // Find the "word" closest to the cursor or selected area
+            // startPos is the cursor location and startPos === endPos when nothing is selected
+
+            // Find the last index of a non-alphanumeric character, dot or minus before the start position
+            const startIndex =
+                startPos > 0
+                    ? this.lastIndexOf(origValue, this.wordBoundryRegex, startPos) + 1
+                    : 0;
+            // Find the first index of a non-alphanumeric character, dot or minus after the start position
+            let endIndex = this.indexOf(origValue, this.wordBoundryRegex, startPos);
+
+            // Set end index to end of string if no matches
+            endIndex = endIndex < 0 ? origValue.length : endIndex;
+
+            // Get the substring that we are acting on
+            replaceText = origValue.substring(startIndex, endIndex);
+
+            // Parse the substring into a number ignoring leading non-numeric characters
+            const origNum = parseInt(replaceText.replace(/^[^\d-]*/, ""));
```

Comment:
We should specify the `radix` because it does not default to 10. See: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/parseInt#parameters

```suggestion
            const originalNumber = parseInt(replaceText.replace(/^[^\d-]*/, ""), 10);
```

Reply:
Doh, I should've remembered that.